### DISCUSSION TITLE
Relocate super search feature within the twin combo boxes

### DISCUSSION THREAD

**Discussion by SJK55** [moneymanagerex/moneymanagerex#5068] [Ideas]
Best to see the image attached. But basically a suggestion that when in the relocate box with the "Relocate" & "to," boxes are displayed,  the fancy search function that exists in other area's is placed within these boxes so that matches can be found if duplication has occurred and you want to change it.

ie unless you know exactly where a category description is 'hiding' this fancy search would allow you to find it.

![Re-alocate request in MMEX](https://user-images.githubusercontent.com/14268231/189516216-3d6568ee-898e-43de-b78c-d4b74c22382a.jpg)

**Comment by whalley** [comment]
Not sure what you mean by 'fancy search' ?? The combobox supports search by typing.

![CleanShot 2022-09-20 at 19 22 26](https://user-images.githubusercontent.com/487345/191334971-0ce16e2c-7478-4328-9965-265764f335d8.png)

Is it this issue? https://github.com/moneymanagerex/moneymanagerex/issues/4892

**Reply by SJK55** [reply]
I have the latest version of MMEX (1.5.21 64 bit)
I click the shopping tag icon up the top of the screen (organize categories)
If I hit enter into the relocate box nothing happens (blank box) I would like it to perform the smart search as it does when doing  a transaction. (ie type any character and it brings up the matches of those characters)

This current box that comes up assumes you have clicked on a top level category to start the search.  It won't work on just typing a subcategory. 

This would be useful if in more than one location the user accidentally placed a subcategory in more than one spot as can happen when you forget that you already created it under Miscellaneous or Bills or Taxes Main Category.

Does that make more sense of what I'm trying to ask for?

I just realized that the click down arrow performs this function. Why can't an Enter on blank do the same thing?
But also if I typed 'Fee' and use down arrow I get a whole lot of unrelated items appearing with no relation to 'Fee' in the text.

Something not right with this function.

**Comment by whalley** [comment]
> If I hit enter into the relocate box nothing happens (blank box) I would like it to perform the smart search as it does when doing a transaction.

OK, agreed that this is not supported. We should perhaps make it more consistent with other category comboboxes. Raised issue https://github.com/moneymanagerex/moneymanagerex/issues/5109

> But also if I typed 'Fee' and use down arrow I get a whole lot of unrelated items appearing with no relation to 'Fee' in the text.

On windows it will find "Feed", "Feel", "Fee:ABC", "Fee"DEF" but not "The Fee" or "Bank:Fees" on windows as the lookup is anchored to the start of the strin. On MacOS it will find all of the above. This is what https://github.com/moneymanagerex/moneymanagerex/issues/4892 covers.

**Comment by whalley** [comment]
@SJK55 Try this beta build - https://ci.appveyor.com/project/moneymanagerex/moneymanagerex/build/job/ujyygixa2dhvwomg/artifacts

**Comment by SJK55** [comment]
Thanks Mark, that does seem to work exactly as I was expecting it to, with the improved combobox consistency.
Excellent!
